rosbag2_transport/test/rosbag2_transport/test_play.cpp

```diff
@@ -174,6 +174,68 @@ TEST_F(RosBag2PlayTestFixture, recorded_messages_are_played_for_all_topics)
           ElementsAre(40.0f, 2.0f, 0.0f)))));
 }
 
+TEST_F(RosBag2PlayTestFixture, recorded_messages_are_played_for_all_topics_multi)
+{
+  auto msg = get_messages_basic_types()[0];
+  msg->int32_value = 42;
+
+  auto topic_types = std::vector<rosbag2_storage::TopicMetadata>{
+    {1u, "topic1", "test_msgs/BasicTypes", "", {}, ""},
+    {2u, "topic2", "test_msgs/BasicTypes", "", {}, ""},
+  };
+
+  // Make sure each reader's/bag's messages are ordered by time
+  // However, do interlace messages across bags
+  constexpr std::size_t num_bags = 3u;
+  std::vector<std::vector<std::shared_ptr<rosbag2_storage::SerializedBagMessage>>> messages_list{};
+  messages_list.emplace_back(std::vector<std::shared_ptr<rosbag2_storage::SerializedBagMessage>>{
+    serialize_test_message("topic1", 1, msg),
+    serialize_test_message("topic2", 5, msg),
+    serialize_test_message("topic1", 8, msg),
+    serialize_test_message("topic2", 10, msg),
+    serialize_test_message("topic1", 13, msg),
+    serialize_test_message("topic2", 14, msg)});
+  messages_list.emplace_back(std::vector<std::shared_ptr<rosbag2_storage::SerializedBagMessage>>{
+    serialize_test_message("topic1", 2, msg),
+    serialize_test_message("topic2", 3, msg),
+    serialize_test_message("topic1", 6, msg),
+    serialize_test_message("topic2", 10, msg),
+    serialize_test_message("topic1", 12, msg),
+    serialize_test_message("topic2", 16, msg)});
+  messages_list.emplace_back(std::vector<std::shared_ptr<rosbag2_storage::SerializedBagMessage>>{
+    serialize_test_message("topic1", 1, msg),
+    serialize_test_message("topic2", 4, msg),
+    serialize_test_message("topic1", 7, msg),
+    serialize_test_message("topic2", 9, msg),
+    serialize_test_message("topic1", 11, msg),
+    serialize_test_message("topic2", 15, msg)});
+  std::vector<rosbag2_transport::Player::reader_storage_options_pair_t> bags{};
+  std::size_t total_messages = 0u;
+  for (std::size_t i = 0u; i < num_bags; i++) {
```

Comment:
```suggestion
  for (std::size_t i = 0u; i < messages_list.size(); i++) {
```

Reply:
changed in fd349cc793eab28feb6ef089c20ad1a6329a4001